src/controllers/problem.ts

```diff
@@ -101,7 +101,7 @@ const problemController = {
                     };
 
 
-                    if (req.body.source && req.body.problemMetadata && req.body.problemMetadata.platformProblemId) {
+                    if (req.body.source && req.body.problemMetadata) {
```

Comment:
See my comment about PUT requests (we won't need this check if we require these fields in validation)